### DISCUSSION TITLE
Using Tailwind CDN works fine but compiled Tailwind or produces limited set of colors.

### DISCUSSION THREAD

**Discussion by drdan18** [tailwindlabs/tailwindcss#6823] [Help] [answered]
When I create a new Laravel 8 app with either Breeze or Jetstream the default dashboard page does not show all of the "expertly-crafted default color palette out-of-the-box" of Tailwind.   I have not altered anything in the tailwindcss.config.js or the webpack.mix.js files.

If I change the head tag to include the CND link instead of the compiled css, all the color classes are working as expected.   
ie.  <script src="https://cdn.tailwindcss.com"></script>

Am I making an incorrect assumption that the entire default color palette should be available when compiled?   

This is what I am expecting and I get it with the CDN:  

![image](https://user-images.githubusercontent.com/6642337/147838302-00ec2797-3cd7-4a5b-8722-2bb5173667a8.png)

This is what I am getting with compiled css: 

![image](https://user-images.githubusercontent.com/6642337/147838275-c5756e5f-c02c-4f11-bc03-5823dab9732c.png)

I am thinking it might be a problem with how TailwindCSS is install on my machine.  I have tried redoing a new project at least 6 times now and always the same result.  Is there a way to confirm the installation of TailwindCSS and/or remove it and reinstall it? 

I am working on a Windows 10 machine using Laragon for development.

**Comment by diahron** [comment]
I am experiencing a similar problem with tailwind when compiling it on my system (Solus-OS...linux distro). I have tried several tutorials for learning Tailwindcss, and getting the same, if not similar results. It will not perform as well as when using the CDN. When using any tutorial version, the HTML will not pick up any of the directives. Output of HTML will remain on page the same as in the editor. If I change the link reference, in the <head> section, to a CDN link everything works fine. I also thought I was doing something simply dumb, but now I see that it is not my imagine or bad typing.

**Reply by drdan18** [reply]
At least I am not alone.   Thanks for posting.

**Comment by DistantMe** [comment]
When you compile, it will scan your source code and only include classes that you're using. Is this what you're seeing, or something else?

**Reply by drdan18** [reply]
Thanks for the reply.   You can see the two pictures I posted above.   They are using the same source code.  All of the default color classes are being called and that is the result I am getting.  All I have done is added this bit of code to the default dashboard page that comes with a new laravel app using breeze.   Maybe I am not scanning this page.  I will check.

**Comment by drdan18** [comment, marked as answer]
Additional testing with the tailwindcss cli.   I ran the following command: 

npx tailwindcss -o ./public/css/app.css

This does create a new app.css file.   My understanding is that it should pull the entire tailwind style sheet which I am assuming should include the default color set.  

It creates the same results that I have shown above where most of the color classes are absent.  If I view this app.css file I can see that only the colors that are showing in the snapshot are the classes that have been added to this file.  

Edit: see reply below for answer.

**Reply by adamwathan** [reply]
Have you configured your “content”?

https://tailwindcss.com/docs/content-configuration

Tailwind 3 only generates the CSS you actually use, so without configuring that none of the CSS will be generated.

**Reply by drdan18** [reply]
Thanks so much for your reply.   It is working now.   I believe I had two errors that were causing the problem.  

I think the original problem was the "content" section of the config.  I did find this part of the docs and made that correction already but it didn't change the output.  It didn't change anything because I made an incorrect test page to generate the color palette visually.

I created it dynamically using inline php in the blade file: 

      @php  $bg_color_class = 'bg-' . strtolower($name) . '-' . $i @endphp
      <td class="p-4 border text-xs text-red-300 {{$bg_color_class}} "> {{$bg_color_class}} </td>

I can understand now that the tailwindcss compiler would not see the classes when they are generated this way, thus they were being purged. 

Thanks again for replying back so quickly.

**Reply by tentree-development** [reply]
I have the same problem only color class generated in bundle.css are the followign is that bug in the latest version? 
`.text-white {
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
}
.text-gray-600 {
  --tw-text-opacity: 1;
  color: rgb(75 85 99 / var(--tw-text-opacity));
}
.text-black {
  --tw-text-opacity: 1;
  color: rgb(0 0 0 / var(--tw-text-opacity));
}
.text-red-500 {
  --tw-text-opacity: 1;
  color: rgb(239 68 68 / var(--tw-text-opacity));
}
.text-green-800 {
  --tw-text-opacity: 1;
  color: rgb(22 101 52 / var(--tw-text-opacity));
}
.text-red-800 {
  --tw-text-opacity: 1;
  color: rgb(153 27 27 / var(--tw-text-opacity));
}
.text-yellow-800 {
  --tw-text-opacity: 1;
  color: rgb(133 77 14 / var(--tw-text-opacity));
}
.text-blue-800 {
  --tw-text-opacity: 1;
  color: rgb(30 64 175 / var(--tw-text-opacity));
}
.text-gray-300 {
  --tw-text-opacity: 1;
  color: rgb(209 213 219 / var(--tw-text-opacity));
}`
